lib/http2.c

```diff
@@ -1146,6 +1146,9 @@ void Curl_http2_done(struct connectdata *conn, bool premature)
   struct HTTP *http = data->req.protop;
   struct http_conn *httpc = &conn->proto.httpc;
 
+  if(!httpc->h2) /* not HTTP/2 ? */
```

Comment:
There is a similar check in ``Curl_http2_setup()``:
```
  if((conn->handler == &Curl_handler_http2_ssl) ||
     (conn->handler == &Curl_handler_http2))
...
```

... not sure which one is better.